kernel/uid_gid.c

```diff
@@ -0,0 +1,775 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#define _GNU_SOURCE
+#include <limits.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <myst/kernel.h>
+#include <myst/syscall.h>
+#include <myst/thread.h>
+
+/* Currently only supports a single mapping which is typically from enclave root
+ * identity to the host application identity when executed. */
+static myst_host_enc_id_mapping host_enc_id_mapping;
+
+void myst_set_host_uid_gid_mapping(myst_host_enc_id_mapping id_mapping)
+{
+    host_enc_id_mapping = id_mapping;
+}
+
+uid_t myst_enc_uid_to_host(uid_t euid)
+{
+    if (euid == host_enc_id_mapping.enc_uid)
+        return host_enc_id_mapping.host_uid;
+    else
+        return euid;
+}
+gid_t myst_enc_gid_to_host(gid_t egid)
+{
+    if (egid == host_enc_id_mapping.enc_gid)
+        return host_enc_id_mapping.host_gid;
+    else
+        return egid;
+}
+
+/* success return 1, fail to read file return -1, not valid user return 0 */
+/* username:password:UID:GID:comment:home directory:default shell */
+static long myst_valid_uid(uid_t uid)
```

Comment:
Maybe add a function to parse the password file and return a list of structures. Then this function can consume the list. The helper function will come in handy when later on we try to get other information from the password file.

Also please rename the function like `myst_valid_uid_based_on_passwd_file`.

Reply:
renamed. 
If it is needed later we can split that out. I dont think returning the whole list is good. If anything we would do what the CRT does which is find an entry based on name or ID and return a structure.

Reply:
The way I read it: we are going parse the password file every time we call `myst_valid_uid` or `myst_valid_gid`.